src/SystemManager.cc

```diff
@@ -112,6 +112,10 @@ size_t SystemManager::ActivatePendingSystems()
     this->systems.push_back(system);
 
     PriorityType p {System::kDefaultPriority};
+    if (system.configurePriority)
+    {
+      p = system.configurePriority->ConfigurePriority();
+    }
```

Comment:
nit: consider logging below if a system has default priority set via the interface but was overriden by the priority in sdf. This would help users to verify the source of truth for their System's priority quickly.

Reply:
added in https://github.com/gazebosim/gz-sim/pull/2500/commits/14118685fabbe95ab3922a524927af191be3a76c; it currently prints a `gzdbg` any time a `<gz:system_priority/>` is read